frontend/src/components/TradeBox/Forms/LightningPayout.tsx

```diff
@@ -145,44 +146,44 @@ export const LightningPayoutForm = ({
     }
   }, [lightning.lnproxyInvoice, lightning.lnproxyAmount]);
 
-  const lnproxyUrl = function () {
-    const bitcoinNetwork = settings?.network ?? 'mainnet';
-    let internetNetwork: 'Clearnet' | 'I2P' | 'TOR' = 'Clearnet';
+  //filter lnproxies when the network settings are updated
+  let bitcoinNetwork: string = 'mainnet';
+  let internetNetwork: 'Clearnet' | 'I2P' | 'TOR' = 'Clearnet';
+  useEffect(() => {
+    bitcoinNetwork = settings?.network ?? 'mainnet';
     if (settings.host?.includes('.i2p')) {
       internetNetwork = 'I2P';
     } else if (settings.host?.includes('.onion') || window.NativeRobosats != undefined) {
       internetNetwork = 'TOR';
     }
 
-    const url = lnproxies[lightning.lnproxyServer][`${bitcoinNetwork}${internetNetwork}`];
-    if (url != 'undefined') {
-      return url;
-    } else {
-      setBadLnproxyServer(
-        t(`Server not available for {{bitcoinNetwork}} bitcoin over {{internetNetwork}}`, {
-          bitcoinNetwork,
+    filteredProxies = lnproxies
+      .filter((node) => node.relayType == internetNetwork)
+      .filter((node) => node.network == bitcoinNetwork);
+  }, [settings.network]);
```

Comment:
Missing dependency `settings.host` . It is best to leave this dependencies array as just `[settings]`, since the parallel work in branch federation-layer adds a variable for "internetNetwork" in settings that will be accessible appwide